DRAFT.md

```diff
@@ -873,161 +888,160 @@ explicit constexpr indirect()
+  type `T` with `*other`, using the allocator `alloc`.
 
 ```c++
 constexpr indirect(allocator_arg_t, const Allocator& a,
                    const indirect& other);
 ```
 
-11. _Mandates_: `T` is a complete type.
+9. _Mandates_: `T` is a complete type.
 
-12. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
+10. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
     valueless, `*this` is valueless. Otherwise, constructs an owned object of
     type `T` with `*other`, using the allocator `alloc`.
 
 ```c++
 constexpr indirect(indirect&& other) noexcept;
 ```
 
-13. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
+11. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
     is `false` then `T` is a complete type.
 
-14. _Effects_: If `other` is valueless, `*this` is valueless. Otherwise, if
-    `alloc == other.alloc` is `true` constructs an object of type `indirect`
-    that owns the owned object of `other`; `other` is valueless. Otherwise,
-    constructs an owned object of type `T` with `*std::move(other)`, using the
-    allocator `alloc`.
+12. _Effects_: `alloc` is direct-non-list initialised from std::move(`other.alloc`).
+  If `other` is valueless, `*this` is valueless. Otherwise `*this` takes
+  ownership of the owned object of `other`.
+
+13. _Postconditions_: `other` is valueless.
 
 ```c++
 constexpr indirect(allocator_arg_t, const Allocator& a, indirect&& other)
-   noexcept(allocator_traits<Allocator>::is_always_equal::value);
+   noexcept(allocator_traits<allocator_type>::is_always_equal::value);
 ```
 
-15. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
+14. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
     is `false` then `T` is a complete type.
 
-16. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
-    valueless, `*this` is valueless. Otherwise, if `alloc == other.alloc` is `true`
+15. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
+    valueless, `*this` is valueless. Otherwise, if `alloc == other.alloc` is `true`,
     constructs an object of type `indirect` that owns the owned object of `other`;
     `other` is valueless. Otherwise, constructs an owned object of type `T` with
     `*std::move(other)`, using the allocator `alloc`.
 
+16. _Postconditions_: `other` is valueless.
+
 ```c++
-template <class... Us>
-explicit constexpr indirect(in_place_t, Us&&... us);
+template <class U=T>
+explicit constexpr indirect(U&& u);
 ```
 
-17. _Constraints_: `is_constructible_v<T, Us...>` is `true`.
-   `is_copy_constructible_v<T>` is `true`.
-   `is_default_constructible_v<allocator_type>` is `true`.
+17. _Constraints_: `is_same_v<remove_cv_ref_t<U>, U>` is `true`.
+    `is_same_v<U, indirect>` is `false`.
+    `is_same_v<U, in_place_t>` is `false`.
```

Comment:
The trait is spelled `remove_cvref_t`, but we don't want this `is_same_v<remove_cvref_t<U>, U>` constraint here at all. We want to allow `U` to be `X&` or `const X&` here. The `is_same_v<remove_cvref_t<U>, U>` constraint was only meant to be added to the ctors of `polymorphic` that take an `inplace_type_t<U>` where it's saying "construct a U", so that we don't allow `in_place_type_t<X&>` which would attempt to store an `X&`.

Here we're always constructing a T, so we don't constrain the U here.

```suggestion
17. _Constraints_: 
    `is_same_v<remove_cvref_t<U>, indirect>` is `false` and
    `is_same_v<remove_cvref_t<U>, in_place_t>` is `false`.
```